### ISSUE TITLE
The `declaration-bang-space-*` fixes write at an index measured before the write in front of it

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#239]
Split out of [#226](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/226), which carries three bugs that share nothing but the review that found them. This is the second of the three.

### What minimal example or steps are needed to reproduce the bug?

A declaration holding more than one `!important` run:

```css
a { b: 1px!important 2px /*c*/!important; }
```

and one holding a double slash, which plain CSS spells no comment with:

```css
a { b: 1px!important//!important; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/declaration-bang-space-before": "always"
  }
}
```

`declaration-bang-space-after` reaches the same shapes from the other side.

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

Every bang of the declaration given the whitespace the option asks for, in one run, with every other character left as it was, and a second run finding nothing left to do.

### What actually happened?

The first fixture comes back as

```css
a { b: 1px !important 2px /*c* /!important; }
```

— the comment's closing delimiter cut in two. **The run after that reports `Unclosed comment` and the file no longer parses.**

The second comes back as `a { b: 1px !important/ /!important; }`, a space written between the two slashes, and does not converge: the run after it gives `a { b: 1px !important/ / !important; }`. It looks like a comment being written into until the index is traced. Nothing here is a comment; plain CSS spells none with a double slash, which is why the shape is reachable at all now that [#216](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/216) has taught the rule to read one.

`declaration-bang-space-after` mis-measures the same input from the other side, and does not converge either: `a { b: 1px! important// !important; }` on the first run and `a { b: 1px! important// ! important; }` on the second.

### What is the cause?

`declarationBangSpaceChecker` searches the whole declaration for `!` and calls `check` for each match with an index counted in the text as it stands. Each fix then writes at its own index — but the write in front of it has already changed the length, so every index past the first is stale.

### Do you have a proposal to fix the bug?

Collect one edit per bang and apply them back to front. `media-feature-range-operator-space-after` already does exactly that: it pushes an index per problem, sorts them descending, and splices.
